src/Servers/HttpSys/src/RequestProcessing/RequestContext.FeatureCollection.cs

```diff
@@ -76,11 +74,39 @@ internal class FeatureContext :
         private bool _responseStarted;
         private bool _completed;
 
-        internal FeatureContext(RequestContext requestContext)
+        internal IFeatureCollection Features => _features; 
+
+        [Flags]
+        // Fields that may be lazy-initialized
+        private enum Fields
+        {
+            None = 0x0,
+            Protocol = 0x1,
+            RequestBody = 0x2,
+            RequestAborted = 0x4,
+            LocalIpAddress = 0x8,
+            RemoteIpAddress = 0x10,
+            LocalPort = 0x20,
+            RemotePort = 0x40,
+            ConnectionId = 0x80,
+            ClientCertificate = 0x100,
+            TraceIdentifier = 0x200,
+        }
+
+        public void InitializeFeatures()
         {
-            _requestContext = requestContext;
+            if (_initialized)
+            {
+                return;
+            }
+
```

Comment:
No longer needed, InitializeFeatures is only called once.
```suggestion
```

Reply:
It's needed for dispose.

Reply:
_initialized is checked in disposal, but this early return isn't needed anymore.

Reply:
Ah right.